### PR TITLE
Added custom ca chain to deployments

### PR DESCRIPTION
**Pull request by b1schumacher**
## Description
Added the custom CA Chain variables to the values.yaml and added the ca to services and jobs.

## Related Issue

- Fixes [ #403](https://github.com/owncloud/ocis-charts/issues/403)

## Motivation and Context
Custom CA chains are required for multiple services, which require that ocis is trusting these ca's

## How Has This Been Tested?

- Tested the deployment with custom CA Chains

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

<!-- or create documentation PR in https://github.com/owncloud/docs ocis/tree/master/modules/ROOT/pages/deployment/container>
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation generated (`make docs`) and committed
- [ ] Documentation ticket raised: <link>
- [x] Documentation PR created: [<link>](https://github.com/owncloud/docs-ocis/pull/871)

### CODE REVIEW

File: charts/ocis/templates/antivirus/deployment.yaml [hunk 1]
```diff
@@ -75,6 +80,10 @@ spec:
             - name: messaging-system-ca
               mountPath: /etc/ocis/messaging-system-ca
               readOnly: true
+            {{- if .Values.customCAChain.enabled }}
+            - name: custom-ca-chain
+              mountPath: /etc/ssl/certs/
+            {{- end }}
```
Comment (d7oc) [inline charts/ocis/templates/antivirus/deployment.yaml:86]: Would be good to have this fragment templated so we don't have these lines in all `deployments.yml` file but only one with a call here.

suggestion

```
{{/*
oCIS ca mount

@param .path          The path to mount
*/}}
{{- define "ocis.caPath" -}}
            {{- if .Values.customCAChain.enabled }}
            - name: custom-ca-chain
              mountPath: {{ .path }}
            {{- end }}
{{- end -}}
```

Usage afterwards with `{{- include "ocis.caPath" /etc/ssl/certs }}`

**Comment by b1schumacher** [comment]
I changed all the entries to a template and fixed also the issue that the customCAChain was overwriting the other CA files. 

Now there are two different directories, one with the default files and one with the custom CA chain. Both are added to the SSL_CERT_DIR variable.

**Comment by wkloucek** [comment]
@b1schumacher I applied some formatting and added enabled this feature in the linting process.

So far I didn't get the pattern for which services the custom CA has been configured / skipped. Could you maybe elaborate why it's not on all services?

Also what's missing are the CronJobs, some of them only talk to NATS but even for this, we could make use of this feature.

**Comment by b1schumacher** [comment]
@wkloucek I added the missing services and the jobs.

**Comment by wkloucek** [comment]
I changed this to point to `stable-5` instead of `main` which is now tracking ocis rolling releases

File: charts/ocis/values.yaml [hunk 2]
```diff
@@ -613,6 +613,15 @@ secretRefs:
...
   globalNotificationsSecretRef: ""
 
+# Custom CA chain
+# If a custom CA chain is needed for the communcation with other services like S3 or notification services.
+# The secret needs to be manually created.
+customCAChain:
+  # -- Custom CA enables SSL_CERT_DIR in pods with the additional path /etc/ssl/custom.
+  enabled: false
+  # -- If custom CA chain is enabled this attribute mounts the existing secret to /etc/ssl/custom.
+  existingSecret: ""
```
Comment (wkloucek) [inline charts/ocis/values.yaml:623]: Is there a particular reason for the CA chain to be put in a secret? From what I know, a CA chain is public information.

Comment (wkloucek) [reply charts/ocis/values.yaml]: I changed it to a ConfigMap in https://github.com/owncloud/ocis-charts/pull/599/commits/e3443826ee3659970b262e7c767d2cac786a4c8d

**Review by wkloucek — APPROVED** [review]

**Review by d7oc — CHANGES_REQUESTED** [review]
LGTM aside the typo. Also verified locally via k3d.

**Review by d7oc — APPROVED** [review]
LGTM

**Pull request merged**